src/cript/api/api.py

```diff
@@ -402,6 +406,791 @@ def api_prefix(self):
+        print(added_uuids)
+
+        # removes["payload_json_removes"] = removed_uuids
+        print("removed_uuids")
+        print(removed_uuids)
+
+        return removes
+
+    ###########################################################################################
+    @staticmethod
+    def generate_patch_removes(dict_obj1, obj2):
+        """
+        dict_obj1 is the object we map
+        obj2 is the object we walk
+        here we get the two objects and take a diff
+        """
+        exclude_regex_paths = [
+            # r"root(\[.*\])?\['uuid'\]",
+            r"root(\[.*\])?\['uid'\]",
+            r"root\['\w+_count'\]",  # All the attributes that end with _count
+            r"root(\[.*\])?\['\w+_count'\]",  # All the attributes that end with _count
+            r"root(\[.*\])?\['locked'\]",
+            r"root(\[.*\])?\['admin'\]",
+            r"root(\[.*\])?\['created_at'\]",
+            r"root(\[.*\])?\['created_by'\]",
+            r"root(\[.*\])?\['updated_at'\]",
+            r"root(\[.*\])?\['updated_by'\]",
+            r"root(\[.*\])?\['public'\]",
+            r"root(\[.*\])?\['notes'\]",
+            r"root(\[.*\])?\['model_version'\]",
+        ]
+
+        # map of first object
+        # uuid_map1 = API.build_uuid_map(obj1)
+        # print("\n\nobj2 ")
+        # print(obj2)
+        # print(type(obj2))
+        # print("\n\nobj2 get json ")
+        # print(obj2.get_json())
+        # print(type(obj2.get_json()))
+        # print("\n\ndict_obj1")
+        # print(dict_obj1)
+        # print(type(dict_obj1))
+
+        print("\n\n********* STARTING EVERYTHING ***********")
+
+        uuid_map1 = API.build_uuid_map(dict_obj1)
+
+        # map 2 has no map because its
+
+        print("********** 2 walking the second obj **********")
+        # walking the second obj
+        for node in obj2:
+            node2_uuid = node.uuid
+
+            if node2_uuid not in uuid_map1:
+                """
+                ---situation where this node is not in map
+                this node: {node2_uuid} was added in a patch to an earlier node so had no comparison
+                """
+                continue
+            else:  # if in uuid
+                """
+                compare "node" and the node from the map
+                node.get_json() and dict_obj1['collection'][0] (for example)
+                """
+
+                path = uuid_map1[node2_uuid]
+
+                # This is getting a node json from the map and object path
+                # So cleaned_original is a node but json
+                cleaned_original = API.get_value_by_path(dict_obj1, path)
+
+                cleaned_original = API.remove_keys_from_dict(cleaned_original)
+
+                modified = json.loads(node.get_json().json)  # Assuming this is already a dictionary
+
+                # cleaned_original = self.remove_keys_from_dict(original)
+                cleaned_modified = API.remove_keys_from_dict(modified)
```

Comment:
better names.

Something like old and new.


Cleaned and orignal don't make that much sense to me.